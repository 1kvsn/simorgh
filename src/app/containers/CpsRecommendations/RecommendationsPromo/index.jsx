import React, { useContext } from 'react';
import pathOr from 'ramda/src/pathOr';
import styled from '@emotion/styled';
import { GEL_SPACING, GEL_SPACING_DBL } from '@bbc/gel-foundations/spacings';
import {
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';
import { getSerifMedium } from '@bbc/psammead-styles/font-styles';
import { getPica } from '@bbc/gel-foundations/typography';
import { grid } from '@bbc/psammead-styles/detection';
import { C_EBON, C_METAL } from '@bbc/psammead-styles/colours';
import { shape, string, oneOfType } from 'prop-types';
import { ServiceContext } from '#contexts/ServiceContext';
import RecommendationsImage from '../RecommendationsPromoImage';
import { storyItem } from '#models/propTypes/storyItem';
import useCombinedClickTrackerHandler from '../../StoryPromo/useCombinedClickTrackerHandler';

const StyledPromoWrapper = styled.div`
  padding: ${GEL_SPACING};
  margin-top: 8px;
  background-color: #f6f6f6;
  > div {
    display: grid;
  }
`;

const PromoGridWrapper = styled.div`
  position: relative;
  @supports (${grid}) {
    display: grid;
    grid-template-columns: repeat(12, 1fr);
    grid-column-gap: ${GEL_SPACING};

    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
      grid-column-gap: ${GEL_SPACING_DBL};
    }

    @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
      display: block;
    }
  }
`;

const ImageGridItem = styled.div`
  display: inline-block;
  position: relative;

  @supports (${grid}) {
    width: initial;
    grid-column: 1 / span 4;
    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
      grid-column: 1 / span 3;
    }
  }
`;

const ImageWrapper = styled.div`
  position: relative;
`;

const TextGridItem = styled.div`
  display: inline-block;

  @supports (${grid}) {
    display: block;
    width: initial;
    padding: initial;
    grid-column: 5 / span 8;
    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
      grid-column: 4 / span 9;
    }
  }
`;

const Link = styled.a`
  position: static;
  color: ${C_EBON};
  text-decoration: none;
  overflow-wrap: break-word;

  &:before {
    bottom: 0;
    content: '';
    left: 0;
    overflow: hidden;
    position: absolute;
    right: 0;
    top: 0;
    white-space: nowrap;
    z-index: 1;
  }

  &:hover,
  &:focus {
    text-decoration: underline;
  }

  &:visited {
    color: ${C_METAL};
  }
`;

const StyledHeadline = styled.h3`
  ${({ service }) => getSerifMedium(service)}
  ${({ script }) => getPica(script)}
  color: ${C_EBON};
  margin: 0;
  padding: ${GEL_SPACING} 0;
`;

const RecommendationsPromo = ({ promo, eventTrackingData }) => {
  const { script, service } = useContext(ServiceContext);
  const handleClickTracking = useCombinedClickTrackerHandler(eventTrackingData);
  const headline = pathOr(null, ['headlines', 'headline'], promo);
  const url = pathOr(null, ['locators', 'assetUri'], promo);
  const indexImage = pathOr(null, ['indexImage'], promo);

  return (
    <StyledPromoWrapper data-e2e="story-promo-wrapper">
      <PromoGridWrapper>
        <ImageGridItem>
          <ImageWrapper>
            <RecommendationsImage indexImage={indexImage} />
          </ImageWrapper>
        </ImageGridItem>
        <TextGridItem>
          <StyledHeadline script={script} service={service}>
            <Link
              href={url}
              onClick={eventTrackingData ? handleClickTracking : null}
            >
              {headline}
            </Link>
          </StyledHeadline>
        </TextGridItem>
      </PromoGridWrapper>
    </StyledPromoWrapper>
  );
};

RecommendationsPromo.propTypes = {
  promo: oneOfType([shape(storyItem)]).isRequired,
  eventTrackingData: shape({
    block: shape({
      componentName: string,
    }),
    link: shape({
      componentName: string,
      url: string,
      format: string,
    }),
  }),
};

RecommendationsPromo.defaultProps = {
  eventTrackingData: null,
};

export default RecommendationsPromo;
