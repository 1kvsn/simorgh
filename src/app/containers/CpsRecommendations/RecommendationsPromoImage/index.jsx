import React from 'react';
import { shape, bool } from 'prop-types';
import pathOr from 'ramda/src/pathOr';
import ImagePlaceholder from '@bbc/psammead-image-placeholder';
import ImageWithPlaceholder from '../../ImageWithPlaceholder';
import { createSrcset } from '#lib/utilities/srcSet';
import getOriginCode from '#lib/utilities/imageSrcHelpers/originCode';
import getLocator from '#lib/utilities/imageSrcHelpers/locator';

const RecommendationsImage = ({ item, lazyLoad }) => {
  const imageValues = pathOr(null, ['indexImage'], item);
  if (!imageValues) {
    const landscapeRatio = (9 / 16) * 100;
    return <ImagePlaceholder ratio={landscapeRatio} />;
  }

  const { height, width, path } = imageValues;

  const ratio = (height / width) * 100;
  const originCode = getOriginCode(path);
  const locator = getLocator(path);
  const imageResolutions = [70, 95, 144, 183, 240, 320, 660];
  const srcset = createSrcset(originCode, locator, width, imageResolutions);
  const sizes = '(max-width: 1008px) 33vw, 321px';
  const DEFAULT_IMAGE_RES = 660;
  const src = `https://ichef.bbci.co.uk/news/${DEFAULT_IMAGE_RES}${path}`;

  return (
    <ImageWithPlaceholder
      alt={imageValues.altText}
      ratio={ratio}
      src={src}
      fallback={false}
      {...imageValues}
      lazyLoad={lazyLoad}
      copyright={imageValues.copyrightHolder}
      srcset={srcset}
      sizes={sizes}
    />
  );
};

RecommendationsImage.propTypes = {
  useLargeImages: bool.isRequired,
  lazyLoad: bool,
  item: shape,
};

RecommendationsImage.defaultProps = {
  lazyLoad: false,
  item: shape({
    imageValues: shape({
      path: '',
      altText: '',
      height: '',
      width: '',
    }),
  }),
};

export default RecommendationsImage;
