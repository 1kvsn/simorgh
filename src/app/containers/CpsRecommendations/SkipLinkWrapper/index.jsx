import React from 'react';
import { node, string } from 'prop-types';
import styled from 'styled-components';
import { visuallyHiddenTextStyle } from '@bbc/psammead-visually-hidden-text';
import { C_EBON, C_WHITE } from '@bbc/psammead-styles/colours';
import { getSansBold } from '@bbc/psammead-styles/font-styles';
import { GEL_BREVIER } from '@bbc/gel-foundations/typography';

const SKIP_LINK_BORDER = '0.125rem';
const GEL_SPACING_PLUS_HALF = `0.75rem`;

const Wrapper = styled.div`
  position: relative;
`;

const SkipLink = styled.a`
  ${({ service }) => getSansBold(service)}
  ${GEL_BREVIER}
  background-color: ${C_WHITE};
  border: ${SKIP_LINK_BORDER} solid ${C_EBON};
  color: ${C_EBON};
  display: block;
  left: 0;
  line-height: 1;
  padding: ${GEL_SPACING_PLUS_HALF};
  position: absolute;
  text-decoration: none;
  top: 0;
  z-index: 10;

  &:not(:focus):not(:active) {
    ${visuallyHiddenTextStyle}
  }
`;

const EndText = styled.p`
  ${visuallyHiddenTextStyle}
`;

const formatSkipText = title => `Skip ${title} and continue reading`;

const formatSkipEndText = title => `End of ${title}`;

const SkipLinkWrapper = ({ service, title, children }) => {
  const endTextId = 'recommendation_skip_link_end_text';
  return (
    <Wrapper>
      <SkipLink service={service} href={`#${endTextId}`}>
        {formatSkipText(title)}
      </SkipLink>
      {children}
      <EndText tabIndex="-1" id={endTextId}>
        {formatSkipEndText(title)}
      </EndText>
    </Wrapper>
  );
};

SkipLinkWrapper.propTypes = {
  service: string.isRequired,
  children: node.isRequired,
  title: string.isRequired,
};

export default SkipLinkWrapper;
