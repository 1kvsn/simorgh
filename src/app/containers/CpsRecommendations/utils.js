const getIchefSrc = href => {
  const DEFAULT_IMAGE_RES = 660;

  const path = href.replace('http://c.files.bbci.co.uk', '/cpsprodpb');

  return `https://ichef.bbci.co.uk/news/${DEFAULT_IMAGE_RES}${path}`;
};

// Transfomrs a list of recommendation items to a story promo format
const transformToStoryPromoItems = items =>
  items.map(({ assetUri, shortHeadline, imageHref }) => ({
    headlines: {
      headline: shortHeadline,
    },
    locators: {
      assetUri,
    },
    indexImage: {
      path: getIchefSrc(imageHref),
      altText: 'Image Alt text', // This value would be changed as it's on our list of improvements.
    },
    uri: assetUri,
  }));

export default transformToStoryPromoItems;
