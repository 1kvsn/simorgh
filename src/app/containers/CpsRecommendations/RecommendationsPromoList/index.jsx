import React, { forwardRef } from 'react';
import { StoryPromoLiBase, StoryPromoUl } from '@bbc/psammead-story-promo-list';
import { arrayOf, shape, string, number } from 'prop-types';
import { storyItem } from '#models/propTypes/storyItem';
import useViewTracker from '#hooks/useViewTracker';
import Grid from '../../../components/Grid';
import RecommendationsPromo from '../RecommendationsPromo';
import getEventTrackingData from './getEventTrackingData';

const RecommendationsPromoListItem = forwardRef(
  ({ item, dir, index }, forwardedRef) => {
    const eventTrackingData = getEventTrackingData({ item, index });
    const linkViewTrackerRef = useViewTracker(eventTrackingData.link);
    const ref = element => {
      linkViewTrackerRef(element);
      forwardedRef(element);
    };

    return (
      <Grid
        item
        ref={ref}
        columns={{
          group0: 1,
          group1: 1,
          group2: 1,
          group3: 1,
          group4: 1,
          group5: 1,
        }}
        as={StoryPromoLiBase}
        border={false}
        key={item.id || item.uri}
      >
        <RecommendationsPromo
          promo={item}
          dir={dir}
          eventTrackingData={eventTrackingData}
        />
      </Grid>
    );
  },
);

const RecommendationsPromoList = ({ promoItems, dir }) => {
  const eventTrackingData = getEventTrackingData();
  const blockViewTrackerRef = useViewTracker(eventTrackingData.block);

  return (
    <Grid
      columns={{
        group0: 1,
        group1: 1,
        group2: 1,
        group3: 1,
        group4: 1,
        group5: 1,
      }}
      as={StoryPromoUl}
      enableGelGutters
    >
      {promoItems.map((item, index) => (
        <RecommendationsPromoListItem
          ref={blockViewTrackerRef}
          index={index}
          item={item}
          dir={dir}
        />
      ))}
    </Grid>
  );
};

RecommendationsPromoListItem.propTypes = {
  dir: string.isRequired,
  item: storyItem.isRequired,
  index: number.isRequired,
};

RecommendationsPromoList.propTypes = {
  dir: string.isRequired,
  promoItems: arrayOf(shape(storyItem)).isRequired,
};

export default RecommendationsPromoList;
