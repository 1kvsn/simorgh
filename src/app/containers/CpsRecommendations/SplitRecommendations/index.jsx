import React from 'react';
import { arrayOf, shape, number, string, func } from 'prop-types';
import { storyItem } from '#models/propTypes/storyItem';
import CpsRecommendations from '..';

/*
 * OPTIMIZELY: 003_hindi_experiment_feature.
 * This component should be removed after the experiment.
 */
const SplitRecommendations = ({ items, ...props }) => {
  const { part, splitRecsViewEventTracker } = props;

  if (!Array.isArray(items)) {
    return null;
  }

  if (part === 1) {
    const firstPart = items && items.slice(0, 2);
    return (
      <CpsRecommendations
        {...props}
        items={firstPart}
        splitRecsViewEventTracker={splitRecsViewEventTracker}
      />
    );
  }

  if (part === 2) {
    const secondPart = items && items.slice(2, 4);
    return (
      <CpsRecommendations
        {...props}
        items={secondPart}
        splitRecsViewEventTracker={splitRecsViewEventTracker}
      />
    );
  }

  return null;
};

SplitRecommendations.propTypes = {
  items: arrayOf(shape(storyItem)),
  showVariation: string.isRequired,
  part: number.isRequired,
  splitRecsViewEventTracker: func.isRequired,
};

SplitRecommendations.defaultProps = {
  items: [],
};

export default SplitRecommendations;
