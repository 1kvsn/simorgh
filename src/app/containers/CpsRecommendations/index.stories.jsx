import React from 'react';
import { storiesOf } from '@storybook/react';

import CpsRecommendations from '.';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ToggleContext } from '#contexts/ToggleContext';

import mundoRecommendationsData from '#pages/StoryPage/fixtureData/recommendations.ltr.json';
import arabicRecommendationsData from '#pages/StoryPage/fixtureData/recommendations.rtl.json';

const renderRecommendations = (items, service) => {
  const toggleState = {
    cpsRecommendations: {
      enabled: true,
    },
  };
  const parentGridCols = {
    group0: 8,
    group1: 8,
    group2: 8,
    group3: 8,
    group4: 8,
    group5: 8,
  };
  return (
    <ServiceContextProvider service={service}>
      <RequestContextProvider
        isAmp={false}
        pageType="STY"
        service={service}
        pathname="/pathname"
      >
        <ToggleContext.Provider
          value={{ toggleState, toggleDispatch: () => {} }}
        >
          <CpsRecommendations items={items} parentColumns={parentGridCols} />
        </ToggleContext.Provider>
      </RequestContextProvider>
    </ServiceContextProvider>
  );
};

storiesOf('Containers|CPS Recommendations/LTR (mundo)', module)
  .addParameters({ chromatic: { disable: true } })
  .add('multiple items', () =>
    renderRecommendations(mundoRecommendationsData, 'mundo'),
  )
  .add('single item', () =>
    renderRecommendations([mundoRecommendationsData[0]], 'mundo'),
  );

storiesOf('Containers|CPS Recommendations/RTL (arabic)', module)
  .addParameters({ chromatic: { disable: true } })
  .add('multiple items', () =>
    renderRecommendations(arabicRecommendationsData, 'arabic'),
  )
  .add('single item', () =>
    renderRecommendations([arabicRecommendationsData[0]], 'arabic'),
  );
