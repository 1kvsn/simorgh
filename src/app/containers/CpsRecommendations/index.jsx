import React, { useContext } from 'react';
import { arrayOf, shape, node } from 'prop-types';
import styled from '@emotion/styled';
import {
  GEL_GROUP_1_SCREEN_WIDTH_MAX,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MAX,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';
import pathOr from 'ramda/src/pathOr';
import path from 'ramda/src/path';
import { C_GHOST } from '@bbc/psammead-styles/colours';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
} from '@bbc/gel-foundations/spacings';
import SectionLabel from '@bbc/psammead-section-label';
import SkipLinkWrapper from '#components/SkipLinkWrapper';

import { storyItem } from '#models/propTypes/storyItem';
import { ServiceContext } from '#contexts/ServiceContext';
import useToggle from '#hooks/useToggle';
import { GridItemMediumNoMargin } from '#app/components/Grid';
import RecommendationsPromo from './RecommendationsPromo';
import RecommendationsPromoList from './RecommendationsPromoList';

const Wrapper = styled.div`
  @media (max-width: ${GEL_GROUP_1_SCREEN_WIDTH_MAX}) {
    padding: 0 ${GEL_SPACING};
  }

  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    padding: 0 ${GEL_SPACING_DBL};
  }
`;

// eslint-disable-next-line react/prop-types
const SkipWrapper = ({ skipLink, service, children }) => {
  return (
    <SkipLinkWrapper service={service} {...skipLink}>
      {children}
    </SkipLinkWrapper>
  );
};

const RecommendationsWrapper = styled.div`
  background-color: ${C_GHOST};
  margin: ${GEL_SPACING_TRPL} 0;
  padding: ${GEL_SPACING_DBL} 0;
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    margin: 0 0 ${GEL_SPACING_TRPL};
    padding: 0 0 ${GEL_SPACING_DBL};
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    margin: 0 0 ${GEL_SPACING_TRPL};
  }
`;

const SingleContentWrapper = styled.div`
  ${({ columnType }) =>
    columnType === 'secondary' &&
    `
    @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
      padding-top: ${GEL_SPACING_DBL};
    }
    @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
      padding-bottom: ${GEL_SPACING_TRPL};
    }
  `}
`;

const LabelComponent = styled(SectionLabel)`
  margin: 0;
  padding: 0;
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    margin: 0 0 ${GEL_SPACING_DBL};
  }
`;

const CpsRecommendations = ({ items }) => {
  const { recommendations, translations, script, service, dir } = useContext(
    ServiceContext,
  );
  const { enabled } = useToggle('cpsRecommendations');
  const labelId = 'recommendations-heading';
  const a11yAttributes = {
    as: 'section',
    role: 'region',
    'aria-labelledby': labelId,
  };

  const hasSingleContent = items.length === 1;
  const [singleContent] = items;

  const CpsOnwardJourneyWrapper = ({ children }) => (
    <Wrapper data-e2e={labelId} {...a11yAttributes}>
      {children}
    </Wrapper>
  );

  CpsOnwardJourneyWrapper.propTypes = {
    children: node.isRequired,
  };

  const { hasStoryRecommendations } = recommendations;

  if (!hasStoryRecommendations || !enabled || !items.length) return null;

  const title = pathOr(
    'You may also be interested in',
    ['recommendationTitle'],
    translations,
  );

  const { text, endTextVisuallyHidden } = path(['skipLink'], recommendations);

  const terms = {
    '%title%': title,
  };

  const endTextId = 'end-of-recommendations';

  const skipLinkProps = {
    endTextId,
    terms,
    text,
    endTextVisuallyHidden,
  };

  return (
    <GridItemMediumNoMargin>
      <RecommendationsWrapper>
        <CpsOnwardJourneyWrapper>
          <SkipWrapper skipLink={skipLinkProps} service={service}>
            {title ? (
              <LabelComponent
                script={script}
                service={service}
                dir={dir}
                labelId={labelId}
                columnType="main"
                overrideHeadingAs="strong"
                bar={false}
                backgroundColor={C_GHOST}
              >
                {title}
              </LabelComponent>
            ) : null}
            {hasSingleContent ? (
              <SingleContentWrapper columnType="main">
                <RecommendationsPromo promo={singleContent} dir={dir} />
              </SingleContentWrapper>
            ) : (
              <RecommendationsPromoList promoItems={items} dir={dir} />
            )}
          </SkipWrapper>
        </CpsOnwardJourneyWrapper>
      </RecommendationsWrapper>
    </GridItemMediumNoMargin>
  );
};

export default CpsRecommendations;

CpsRecommendations.propTypes = {
  items: arrayOf(shape(storyItem)),
};

CpsRecommendations.defaultProps = {
  items: [],
};
