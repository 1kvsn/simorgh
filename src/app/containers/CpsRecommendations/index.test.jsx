import React from 'react';
import { render } from '@testing-library/react';
import { latinDiacritics } from '@bbc/gel-foundations/scripts';
import { ServiceContext } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ToggleContext } from '#contexts/ToggleContext';

import recommendationsData from '#pages/StoryPage/fixtureData/recommendations.ltr.json';

import CpsRecommendations from '.';

const parentColumns = {
  group0: 8,
  group1: 8,
  group2: 8,
  group3: 8,
  group4: 12,
  group5: 12,
};

const renderContainer = (items, hasStoryRecommendations, toggleEnabled) => {
  const toggleState = {
    cpsRecommendations: {
      enabled: toggleEnabled,
    },
  };
  return render(
    <ServiceContext.Provider
      value={{
        script: latinDiacritics,
        service: 'mundo',
        dir: 'ltr',
        recommendations: {
          hasStoryRecommendations,
        },
      }}
    >
      <RequestContextProvider
        bbcOrigin="https://www.test.bbc.co.uk"
        isAmp={false}
        pageType="STY"
        pathname="/mundo/085965"
        service="mundo"
        statusCode={200}
      >
        <ToggleContext.Provider
          value={{ toggleState, toggleDispatch: jest.fn() }}
        >
          <CpsRecommendations items={items} parentColumns={parentColumns} />
        </ToggleContext.Provider>
      </RequestContextProvider>
    </ServiceContext.Provider>,
  );
};

describe('CpsRecommendations', () => {
  it('should not render when cpsRecommendations toggle is disabled', () => {
    const toggleEnabled = false;
    const hasStoryRecommendations = true;

    const { container } = renderContainer(
      recommendationsData,
      hasStoryRecommendations,
      toggleEnabled,
    );
    expect(container).toMatchSnapshot();
  });
  it('should not render when the hasStoryRecommendations flag is disabled for the service', () => {
    const hasStoryRecommendations = false;
    const toggleEnabled = true;

    const { container } = renderContainer(
      recommendationsData,
      hasStoryRecommendations,
      toggleEnabled,
    );
    expect(container).toMatchSnapshot();
  });

  it('should not render when there is no recommendations data', () => {
    const hasStoryRecommendations = false;
    const toggleEnabled = true;

    const { container } = renderContainer(
      [],
      hasStoryRecommendations,
      toggleEnabled,
    );
    expect(container).toMatchSnapshot();
  });

  describe('should render when cpsRecommendations toggle and hasStoryRecommendations flag are both true', () => {
    const toggleEnabled = true;
    const hasStoryRecommendations = true;

    it('for multiple items', () => {
      const { container } = renderContainer(
        recommendationsData,
        hasStoryRecommendations,
        toggleEnabled,
      );
      expect(container).toMatchSnapshot();
    });
    it('for a single item', () => {
      const { container } = renderContainer(
        [recommendationsData[0]],
        hasStoryRecommendations,
        toggleEnabled,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
