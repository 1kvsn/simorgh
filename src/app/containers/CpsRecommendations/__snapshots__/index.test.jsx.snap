// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CpsRecommendations should not render when cpsRecommendations toggle is disabled 1`] = `<div />`;

exports[`CpsRecommendations should not render when the hasStoryRecommendations flag is disabled for the service 1`] = `<div />`;

exports[`CpsRecommendations should not render when there is no recommendations data 1`] = `<div />`;

exports[`CpsRecommendations should render when cpsRecommendations toggle and hasStoryRecommendations flag are both true for a single item 1`] = `
<div>
  <div
    class="RecommendationsWrapper-sc-1p91yk0-1 isuewT"
  >
    <section
      aria-labelledby="recommendations-heading"
      class="Wrapper-opyob2-0 dlivvF"
      data-e2e="related-content"
      role="region"
    >
      <div
        class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD StyledSectionLabel-opyob2-2 dfvtmF"
      >
        <strong
          class="Heading-sc-1x4gwjz-2 eZofTG"
        >
          <span
            class="FlexColumn-wx5kdp-0 dgpsiS"
          >
            <span
              class="FlexRow-wx5kdp-2 dVBNyZ"
            >
              <span
                class="Title-wx5kdp-4 lmzCD"
                dir="ltr"
                id="recommendations-heading"
              >
                Recommendations
              </span>
            </span>
          </span>
        </strong>
      </div>
      <div
        class="SingleContentWrapper-opyob2-3 dFLSlx"
      >
        <div
          class="GridComponent-nf79gm-0 elxESl"
          dir="ltr"
        >
          <div
            class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 lgUEsJ"
          >
            <div
              class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
              dir="ltr"
            >
              <div
                class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
              >
                <img
                  alt="Angie Dodge"
                  class="StyledImg-sc-7vx2mr-0 hIxkbt"
                  src="https://ichef.bbci.co.uk/news/660/cpsprodpb/BD47/production/_111155484_mediaitem111155481.jpg"
                />
              </div>
            </div>
            <div
              class="TextGridItem-sc-1lkbq5i-0 jbyRcL"
              dir="ltr"
            >
              <span
                class="Headline-sc-1dvfmi3-3 dnWhDh"
              >
                <a
                  class="Link-sc-1dvfmi3-5 jeUMCT"
                  href="/mundo/noticias-53377054"
                >
                  La conmovedora historia de cómo una madre y el hombre preso por la muerte de su hija se unieron para atrapar al verdadero asesino
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`CpsRecommendations should render when cpsRecommendations toggle and hasStoryRecommendations flag are both true for multiple items 1`] = `
<div>
  <div
    class="RecommendationsWrapper-sc-1p91yk0-1 isuewT"
  >
    <section
      aria-labelledby="recommendations-heading"
      class="Wrapper-opyob2-0 dlivvF"
      data-e2e="related-content"
      role="region"
    >
      <div
        class="SectionLabelWrapper-sc-1x4gwjz-1 hsirkD StyledSectionLabel-opyob2-2 dfvtmF"
      >
        <strong
          class="Heading-sc-1x4gwjz-2 eZofTG"
        >
          <span
            class="FlexColumn-wx5kdp-0 dgpsiS"
          >
            <span
              class="FlexRow-wx5kdp-2 dVBNyZ"
            >
              <span
                class="Title-wx5kdp-4 lmzCD"
                dir="ltr"
                id="recommendations-heading"
              >
                Recommendations
              </span>
            </span>
          </span>
        </strong>
      </div>
      <ul
        class="StoryPromoUl-sc-171lqjd-2 brjIMO GridComponent-nf79gm-0 buIDdq"
        dir="ltr"
        role="list"
      >
        <li
          class="StoryPromoLiBase-sc-171lqjd-0 kPpUUd GridComponent-nf79gm-0 cjfxRv"
          dir="ltr"
          role="listitem"
        >
          <div
            class="GridComponent-nf79gm-0 eRYAGr"
            dir="ltr"
          >
            <div
              class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 lgUEsJ"
            >
              <div
                class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
                dir="ltr"
              >
                <div
                  class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
                >
                  <img
                    alt="Angie Dodge"
                    class="StyledImg-sc-7vx2mr-0 hIxkbt"
                    src="https://ichef.bbci.co.uk/news/660/cpsprodpb/BD47/production/_111155484_mediaitem111155481.jpg"
                  />
                </div>
              </div>
              <div
                class="TextGridItem-sc-1lkbq5i-0 jbyRcL"
                dir="ltr"
              >
                <span
                  class="Headline-sc-1dvfmi3-3 dnWhDh"
                >
                  <a
                    class="Link-sc-1dvfmi3-5 jeUMCT"
                    href="/mundo/noticias-53377054"
                  >
                    La conmovedora historia de cómo una madre y el hombre preso por la muerte de su hija se unieron para atrapar al verdadero asesino
                  </a>
                </span>
              </div>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLiBase-sc-171lqjd-0 kPpUUd GridComponent-nf79gm-0 cjfxRv"
          dir="ltr"
          role="listitem"
        >
          <div
            class="GridComponent-nf79gm-0 eRYAGr"
            dir="ltr"
          >
            <div
              class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 lgUEsJ"
            >
              <div
                class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
                dir="ltr"
              >
                <div
                  class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
                >
                  <img
                    alt="Edy Fonseca"
                    class="StyledImg-sc-7vx2mr-0 hIxkbt"
                    src="https://ichef.bbci.co.uk/news/660/cpsprodpb/11DDC/production/_112608137_dsc04722.jpg"
                  />
                </div>
              </div>
              <div
                class="TextGridItem-sc-1lkbq5i-0 jbyRcL"
                dir="ltr"
              >
                <span
                  class="Headline-sc-1dvfmi3-3 dnWhDh"
                >
                  <a
                    class="Link-sc-1dvfmi3-5 jeUMCT"
                    href="/mundo/noticias-america-latina-52884902"
                  >
                    Coronavirus en Colombia | El caso de la vigilante Edy Fonseca: "Me retuvieron en un edificio de clase alta durante la pandemia, lo acepté y me siento culpable"
                  </a>
                </span>
              </div>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLiBase-sc-171lqjd-0 kPpUUd GridComponent-nf79gm-0 cjfxRv"
          dir="ltr"
          role="listitem"
        >
          <div
            class="GridComponent-nf79gm-0 eRYAGr"
            dir="ltr"
          >
            <div
              class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 lgUEsJ"
            >
              <div
                class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
                dir="ltr"
              >
                <div
                  class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
                >
                  <img
                    alt="Martha Mukamushinzimana dice que sólo estaba siguiendo órdenes."
                    class="StyledImg-sc-7vx2mr-0 hIxkbt"
                    src="https://ichef.bbci.co.uk/news/660/cpsprodpb/0B66/production/_112981920_53113381.jpg"
                  />
                </div>
              </div>
              <div
                class="TextGridItem-sc-1lkbq5i-0 jbyRcL"
                dir="ltr"
              >
                <span
                  class="Headline-sc-1dvfmi3-3 dnWhDh"
                >
                  <a
                    class="Link-sc-1dvfmi3-5 jeUMCT"
                    href="/mundo/noticias-internacional-53113381"
                  >
                    Soy una mujer genocida y aún me persiguen los recuerdos de lo que hice
                  </a>
                </span>
              </div>
            </div>
          </div>
        </li>
        <li
          class="StoryPromoLiBase-sc-171lqjd-0 kPpUUd GridComponent-nf79gm-0 cjfxRv"
          dir="ltr"
          role="listitem"
        >
          <div
            class="GridComponent-nf79gm-0 eRYAGr"
            dir="ltr"
          >
            <div
              class="StoryPromoWrapper-sc-1dvfmi3-0 buBJys StoryPromoWrapper-sc-1p91yk0-0 lgUEsJ"
            >
              <div
                class="ImageGridItem-sc-1xk56cj-0 fNYKlW"
                dir="ltr"
              >
                <div
                  class="ImageContentsWrapper-sc-1dvfmi3-1 cinwNl"
                >
                  <img
                    alt="Julia Garner en The Assitant"
                    class="StyledImg-sc-7vx2mr-0 hIxkbt"
                    src="https://ichef.bbci.co.uk/news/660/cpsprodpb/98E7/production/_113334193_10f06dbb-371a-4aa0-9577-7152063e5f97.jpg"
                  />
                </div>
              </div>
              <div
                class="TextGridItem-sc-1lkbq5i-0 jbyRcL"
                dir="ltr"
              >
                <span
                  class="Headline-sc-1dvfmi3-3 dnWhDh"
                >
                  <a
                    class="Link-sc-1dvfmi3-5 jeUMCT"
                    href="/mundo/vert-cul-53370542"
                  >
                    Estrenos 2020: las 10 mejores películas de lo que va del año según la BBC (y que seguramente no pudiste ver en el cine)
                  </a>
                </span>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </section>
  </div>
</div>
`;
